contract/src/stage1/function_call.rs

```diff
@@ -66,23 +92,29 @@ impl DropZone {
 
         near_sdk::log!(
             "Attaching Total: {:?} Deposit: {:?} Should Refund?: {:?} Amount To Refund: {:?} With args: {:?}", 
-            yocto_to_near(fc_data.deposit.0 + if fc_data.refund_to_deposit.unwrap_or(false) {amount_to_refund} else {0}),
-            yocto_to_near(fc_data.deposit.0), fc_data.refund_to_deposit.unwrap_or(false), yocto_to_near(amount_to_refund),
+            yocto_to_near(method_data.deposit.0 + if fc_config.clone().and_then(|c| c.refund_to_deposit).unwrap_or(false) {amount_to_refund} else {0}),
+            yocto_to_near(method_data.deposit.0), fc_config.clone().and_then(|c| c.refund_to_deposit).unwrap_or(false), yocto_to_near(amount_to_refund),
             final_args
         );
 
         // Call function with the min GAS and deposit. all unspent GAS will be added on top
-        Promise::new(fc_data.receiver).function_call_weight(
-            fc_data.method,
+        Promise::new(method_data.receiver.clone()).function_call_weight(
+            method_data.method.clone(),
             final_args.as_bytes().to_vec(),
             // The claim is successful so attach the amount to refund to the deposit instead of refunding the funder.
-            fc_data.deposit.0
-                + if fc_data.refund_to_deposit.unwrap_or(false) {
+            method_data.deposit.0
+                + if fc_config
+                    .clone()
+                    .and_then(|c| c.refund_to_deposit)
+                    .unwrap_or(false)
+                {
                     amount_to_refund
                 } else {
                     0
                 },
-            fc_data.gas_if_straight_execute.unwrap_or(Gas(0)),
+            fc_config
+                .and_then(|c| c.gas_if_straight_execute)
```

Comment:
just a nit, we might want to rename this

"straight" is a bit confusing. How about `gas_if_claim_only` because that's more what it is?

Could just find and replace later

Reply:
Agree

Reply:
Will fix in another PR